.github/workflows/integration-tests.yml

```diff
@@ -1541,9 +1543,13 @@ jobs:
                     path: cloudgatewayservicerouting
 
             -   name: Code coverage and publish results
-                run: >
-                    ./gradlew --info coverage sonar -Dresults="containercitests/results,citestswithinfinispan/results,containercitestszosmfrsu2012/results,ContainerCITestsWithRedisReplica/results,ContainerCITestsWithRedisSentinel/results,containercitestsinternalport/results,cloudgatewayproxy/results,citestswebsocketchaoticha/results,cloudgatewayservicerouting/results,containercitestszaas/results"
-                    -Psonar.host.url=$SONAR_HOST_URL -Dsonar.token=$SONAR_TOKEN -Partifactory_user=$ARTIFACTORY_USERNAME -Partifactory_password=$ARTIFACTORY_PASSWORD
+                run: |
+                    export JAVA_HOME=$JAVA_HOME_8_${{ runner.arch }}
+                    echo Execute Code coverage task using Java 8: JAVA_HOME = $JAVA_HOME
+                    ./gradlew --info coverage -Dresults="containercitests/results,citestswithinfinispan/results,containercitestszosmfrsu2012/results,ContainerCITestsWithRedisReplica/results,ContainerCITestsWithRedisSentinel/results,containercitestsinternalport/results,cloudgatewayproxy/results,citestswebsocketchaoticha/results,cloudgatewayservicerouting/results,containercitestszaas/results"
+                    export JAVA_HOME=$JAVA_HOME_17_${{ runner.arch }}
+                    echo Execute Sonar scanner using Java 17: JAVA_HOME = $JAVA_HOME
+                    ./gradlew --info sonar -Psonar.host.url=$SONAR_HOST_URL -Dsonar.token=$SONAR_TOKEN -Partifactory_user=$ARTIFACTORY_USERNAME -Partifactory_password=$ARTIFACTORY_PASSWORD
```

Comment:
is there a reason why both can't run with java 17?

Reply:
Because coverage task triggers the whole build and test ... and with Java 17 tests are difficult (that was resolved in v3). This way the sonar task will pick up the results for cache and runs in with desired Jave 17.

Reply:
got it, thank you